app/components/pipeline/workflow/component.js

```diff
@@ -143,6 +147,18 @@ export default class PipelineWorkflowComponent extends Component {
         this.event.id,
         this.buildsCallback
       );
+
+      const { workflowGraph } = this.event;
+      const eventStages = extractEventStages(workflowGraph);
+
+      if (eventStages && eventStages.length) {
```

Comment:
```suggestion
      if (this.stages.length > 0) {
```

Reply:
Event stages is a snapshot of pipeline stages when the event was created. We don't need to load/reload stage builds if an event was created when the pipeline did not have stages configured.

So we cannot use pipeline stages here.

Reply:
That's fine.  There's other issues with the event snapshot to be addressed at a later time.

`extractEventStages` returns an array so just evaluate the length.
```suggestion
      const { workflowGraph } = this.event;
      const eventStages = extractEventStages(workflowGraph);

      if (eventStages.length) {
```